CharlieBackend.AdminPanel/Services/CalendarService.cs

```diff
@@ -0,0 +1,152 @@
+﻿using AutoMapper;
+using CharlieBackend.AdminPanel.Models.Calendar;
+using CharlieBackend.AdminPanel.Services.Interfaces;
+using CharlieBackend.AdminPanel.Utils.Interfaces;
+using CharlieBackend.Core.DTO.Course;
+using CharlieBackend.Core.DTO.Mentor;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Core.DTO.Student;
+using CharlieBackend.Core.DTO.StudentGroups;
+using CharlieBackend.Core.DTO.Theme;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.AdminPanel.Services
+{
+    /// <summary>
+    /// Class that provides methods for work with calendar and it's data.
+    /// </summary>
+    public class CalendarService : ICalendarService
+    {
+        private readonly IScheduleService _scheduleService;
+        private readonly IApiUtil _apiUtil;
+        private readonly IMapper _mapper;
+
+        private const string _getActiveCoursesEndpoint = "api/courses/isActive";
+        private const string _getActiveMetorsEndpoint = "api/mentors/active";
+        private const string _getStudentGroupsEndpoint = "api/student_groups";
+        private const string _getActiveStudentsEndpoint = "api/students/active";
+        private const string _getThemesEndpoint = "api/themes";
+
+        private const int defaultDateFilterOffset = 15;
+
+        public CalendarService(
+            IScheduleService scheduleService,
+            IApiUtil apiUtil,
+            IMapper mapper)
+        {
+            _scheduleService = scheduleService;
+            _apiUtil = apiUtil;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Makes parallel asynchronous HTTP calls to API service and returns obtained data in a form
+        /// of a ViewModel that is ready to be viewed in a calendar.
+        /// </summary>
+        /// <param name="scheduledEventFilter">DTO with an optional set of 
+        /// filters for scheduled events.</param>
+        /// <returns>>model with data required for viewing a calendar.</returns>
+        public async Task<CalendarViewModel> GetCalendarDataAsync(
+            ScheduledEventFilterRequestDTO scheduledEventFilter)
+        {
+            ApplyDefaultDateEventFilter(scheduledEventFilter);
+
+            var getCoursesTask = Task.Run(() => GetActiveCourseViewModelsAsync());
```

Comment:
GetActiveCourseViewModelsAsync is already a Task, so you can skip Task.Run()

Reply:
I'm sorry I merged this pull request too soon.
Removing redundant `Task.Run()` calls in https://github.com/ita-social-projects/WhatBackend/pull/483